src/DeviceInterface/tests/initFocusApi.test.js

```diff
@@ -0,0 +1,318 @@
+        settings.featureToggles.input_focus_api = false;
+
+        // Reinitialize the API because the feature toggle has changed
+        focusApi.cleanup();
+        focusApi = initFocusApi(forms, settings, attachKeyboardCallback);
+
+        attachAndReturnGenericForm(`
+            <form>
+                <input type="email" name="email" id="email-input" />
+            </form>
+        `);
+
+        const emailInput = document.getElementById('email-input');
+
+        const focusEvent = new FocusEvent('focus', { bubbles: true });
+        Object.defineProperty(focusEvent, 'target', { value: emailInput, enumerable: true });
+        window.dispatchEvent(focusEvent);
+
+        expect(attachKeyboardCallback).not.toHaveBeenCalled();
+    });
+
+    it('should set autocomplete attribute on email input when autocomplete_attribute_support is enabled', () => {
+        attachAndReturnGenericForm(`
+            <form>
+                <input type="email" name="email" id="email-input" />
+            </form>
+        `);
+
+        const emailInput = /** @type {HTMLInputElement} */ (document.getElementById('email-input'));
+
+        emailInput.setAttribute('data-ddg-inputType', 'identities.emailAddress');
+
+        const focusEvent = new FocusEvent('focus', { bubbles: true });
+        Object.defineProperty(focusEvent, 'target', { value: emailInput, enumerable: true });
+        window.dispatchEvent(focusEvent);
+
+        expect(emailInput.getAttribute('autocomplete')).toBe('email');
+    });
+
+    it('should set autocomplete attribute on name input fields', () => {
+        attachAndReturnGenericForm(`
+            <form>
+                <input type="text" name="firstName" id="first-name" placeholder="First Name" />
+                <input type="text" name="lastName" id="last-name" placeholder="Last Name" />
+                <input type="text" name="fullName" id="full-name" placeholder="Full Name" />
+            </form>
+        `);
+
+        const firstNameInput = /** @type {HTMLInputElement} */ (document.getElementById('first-name'));
+        const lastNameInput = /** @type {HTMLInputElement} */ (document.getElementById('last-name'));
+        const fullNameInput = /** @type {HTMLInputElement} */ (document.getElementById('full-name'));
+
+        firstNameInput.setAttribute('data-ddg-inputType', 'identities.firstName');
+        lastNameInput.setAttribute('data-ddg-inputType', 'identities.lastName');
+        fullNameInput.setAttribute('data-ddg-inputType', 'identities.fullName');
+
+        firstNameInput.focus();
+        expect(firstNameInput.getAttribute('autocomplete')).toBe('given-name');
+
+        lastNameInput.focus();
+        expect(lastNameInput.getAttribute('autocomplete')).toBe('family-name');
+
+        fullNameInput.focus();
+        expect(fullNameInput.getAttribute('autocomplete')).toBe('name');
+    });
+
+    it('should set autocomplete attribute on phone input', () => {
```

Comment:
Not sure why you went for separate tests for each individual input type. Just makes the whole test file larger and harder to follow. You could have just a single form with all fields, loop over the fields, and the do the `.focus(); expect()…` for each. Or even without the loop since you still have to define the expected values manually, so a loop doesn't really save you much. Anyway.